woocommerce/woocommerce-ios
# Shipment Tracking Mark 2: Storage framework updates

This PR primarily contains the updates to the v10 core data model in the `Storage` framework. Since v10 was already a heavyweight migration because of #671, I ended up regenerating the v9→v10 mapping file (and also renaming it to `WooCommerceModelV9toV10.xcmappingmodel`).

This new mapping model now accounts for a brand new entity: `ShipmentTracking`. In addition, I also added some of the boilerplate Yosemite framework things like `ReadOnlyType` & `ReadOnlyConvertible` implementations.

Finally I gave `CoreDataIterativeMigrator.swift` a tiny bit of formatting ❤️ so it follows our project standards more — no logic was added nor taken away.

Ref: #250 

## Testing

I would start by running the unit tests and making sure they are ✅ 

Next, since this PR really just contains updates to the v10 data model + migration, we can follow the test instructions @astralbodies provided in #671:

1. Switch back to the `release/1.2` branch. Make sure your target does not have any version of the Woo app installed...then Clean, rebuild, and launch.
2. Sign in, make sure to fetch orders.
3. Kill the app
4. Close Xcode, switch to this branch.
5. Reload Xcode, clean, rebuild, and launch.
    - [ ] Verify the app loads and is still logged into the store you previously loaded ☝️ 
6. Switch to orders tab - orders should already be present

@astralbodies and @mindgraffiti  would you mind giving this a look/test?
